### PR TITLE
Fix comments

### LINKED ISSUE #332: BUG: Visitors should not be allowed to alter comments
### Is there an existing issue raised for this?

- [X] I have searched the existing issues

### Describe your issue

Users visiting the website cannot delete other persons comments. A comment can only be deleted by the user that wrote it.
If the comments are hardcoded I would like to implement real backend for users to comment.

### Steps to reproduce

go to /recipe/:id
head to the comment section
click on the delele button 
The comment gets deleted.

### What was the expected result?

a user writing the comment should be allowed to delete.

### Proposed solution

I will create the necessary APIs for the purpose.

### Would you like to work on this issue?

Yes

### Terms & Conditions

- [X] I agree to follow the Code of Conduct
- [X] I'm a GSSOC'24 Ext contributor

### PR DESCRIPTION
**Pull request by haseebzaki-07**
## What does this PR do?

This PR updates the comments schema, controller for creating and deleting comments also modifies the frontend API calls to match the purpose.

Screenshot 2024-10-30 152420
Screenshot 2024-10-30 152642

https://github.com/user-attachments/assets/e5a94226-5971-431a-ac45-fe7b32ab443b

Fixes #332 

## How should this be tested?

go to the recipe/:id page and try to delete the comments of other users.

### Required

- [X] Filled out the "How to test" section in this PR
- [X] Read Contributing guide
- [X] Self-reviewed my own code
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] The changes don't cause any responsiveness issues

### CODE REVIEW

**Comment by AlfiyaSiddique** [comment]
@haseebzaki-07 the main aim of the issue is to stop a user from deleting others' comments, which is not shown in the video.
Either for users who are not allowed to delete comments do not show the button or on clicking the delete button raise toast saying you are not eligible for this action!

Also your seed data code is in this PR make sure to remove it

**Comment by haseebzaki-07** [comment]
@AlfiyaSiddique I will add the video from other people perspective and do the changes.

**Comment by haseebzaki-07** [comment]
hey @AlfiyaSiddique done the changes and video from others perspective
https://github.com/user-attachments/assets/3939afec-952d-4b91-b69a-8c613e6287cf

File: backend/models/Comment.js [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
+    default: Date.now,
+  },
 });
 
-const Comment = mongoose.model('Comment', commentSchema);
+const Comment = mongoose.model("Comment", commentSchema);
 
-export default Comment;
\ No newline at end of file
+export default Comment;
```
Comment (AlfiyaSiddique) [inline backend/models/Comment.js:33]: @haseebzaki-07 make sure the change in model does not change any other functionality

File: frontend/src/Pages/Recipes.jsx [hunk 2]
```diff
@@ -44,8 +44,12 @@ const Recipes = ({ type }) => {
                 },
             });
 
+            console.log("Response data:", response.data); // Add this line
+
```
Comment (AlfiyaSiddique) [inline frontend/src/Pages/Recipes.jsx:48]: @haseebzaki-07 remove this log

File: backend/models/Comment.js (hunk 1 above)
Comment (haseebzaki-07) [reply backend/models/Comment.js:33]: @AlfiyaSiddique the model change does not affect any other functionality.

**Comment by AlfiyaSiddique** [comment]
@haseebzaki-07 some comments and make sure that the parameters are passed properly from frontend because getting this error in console:
image

**Comment by haseebzaki-07** [comment]
hey @AlfiyaSiddique I fixed the error logs and there are no problems form the frontend

**Review by AlfiyaSiddique — APPROVED** [review]

**Pull request merged**